### PR TITLE
Enabling compatibility with require for loading d3

### LINKED ISSUE #1525: Correctly add d3 when require.js is present
Currently Plottable expects to find d3 on window but require.js prevents this from happening.  If require is present then need to call require("d3").

### PR DESCRIPTION
**Pull request by bluong**
Fixes #1525

### CODE REVIEW

File: src/reference.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 /// <reference path="../bower_components/svg-typewriter/svgtypewriter.d.ts" />
 
+/// <reference path="core/init.ts" />
```
Comment (jtlan) [inline src/reference.ts]: Does this guarantee that **init.ts** is run first?

Comment (bluong) [inline src/reference.ts]: I thought reference.ts defines the order in which the typescript files are compiled / built.  Is this incorrect?

**Comment by jtlan** [comment]
Let's not do this -- it doesn't make sense for us to build a feature to work with a version of D3 we don't support.

**Comment by bluong** [comment]
As of 3.4.10, d3 is always registered in the global namespace, so this isn't really needed. https://github.com/mbostock/d3/pull/1921

**Pull request closed without merging**